dsp/window/window.go

```diff
@@ -330,6 +330,57 @@ func (g Gaussian) Transform(seq []float64) []float64 {
 	return seq
 }
 
+// the values for specrtal leakage come from
+// A.D. Poularikas, "The Handbook of Formulas and Tables for Signal Processing"
+// table 7.1
+
+// Tukey can modify a sequence by the Tukey window and return the result.
+// See https://en.wikipedia.org/wiki/Window_function#Tukey_window
+// and https://prod-ng.sandia.gov/techlib-noauth/access-control.cgi/2017/174042.pdf page 88
+//
+// The Tukey window is an adjustible window. It can be thought of as something
+// between a rectangular and a Hann window, with a flat center and tapered edges.
+//
+// The properties of the window depend on the value of α (alpha). α controls
+// the fraction of the window which contains a cosine taper. α = 0.5 gives a
+// window whos central 50% is a flat top and outer quartiles are tapered.
+// 0 < α < 1; if α is outside the bounds, it is treated as 0 or 1.
+//
+// Spectral leakage parameters for Alpha=0.5 are
+// ΔF_0 = 1.22, ΔF_0.5 = 1.15, K = 1.3, ɣ_max = -15, β = -2.5.
+type Tukey struct {
+	Alpha float64
+}
+
+// Transform applies the Tukey transformation to seq in place, using the value
+// of the receiver as the Alpha parameter, and returning the result
+func (t Tukey) Transform(seq []float64) []float64 {
+	if t.Alpha <= 0 {
+		return Rectangular(seq)
+	} else if t.Alpha >= 1 {
+		return Hann(seq)
+	}
+
+	sep := 0.5 * t.Alpha * float64(len(seq)-1)
+	sepI := int(math.Floor(sep)) + 1
+	sep2 := float64((len(seq) - 1)) * (1 - .5*t.Alpha)
+	sep2I := int(math.Ceil(sep2))
+	for i := range seq {
```

Comment:
Some opportunities for optimisation here, to avoid `if` statements inside the `for` loop.

Reply:
Yes, see mine [here](https://github.com/gonum/gonum/pull/1437#discussion_r484200077).